Tasks/AzureRmWebAppDeployment/Utility.ps1

```diff
@@ -2,10 +2,26 @@ $ErrorActionPreference = 'Stop'
 
 function Get-MsDeployExePath
 {
-    $currentDir = (Get-Item -Path ".\").FullName
+    $msDeployExePath = $null
+    try
+    {
+        $msDeployExePath = Get-MSDeployOnTargetMachine
+    }
+    catch [System.Exception]
+    {
+         Write-Verbose ("MSDeploy is not installed in system." + $_.Exception.Message)
+    }
+
+    if( $msDeployExePath -eq $null )
+    {
 
-    $msDeployExeDir = Join-Path $currentDir "MSDeploy3.6"
-    $msDeployExePath = Join-Path $msDeployExeDir "msdeploy.exe"
+        Write-Verbose  (Get-LocalizedString -Key "Using local MSDeploy.exe")  
+        $currentDir = (Get-Item -Path ".\").FullName
+        $msDeployExeDir = Join-Path $currentDir "MSDeploy3.6"
```

Comment:
doesn't it vary based on the version?
If we support only this version - pass this as a parameter to the Get-MSDeployOnTargetMachine cmdlet